packages/react-dom-bindings/src/server/ReactDOMServerExternalRuntime.js

```diff
@@ -13,13 +14,91 @@ import {
   completeSegment,
 } from './fizz-instruction-set/ReactDOMFizzInstructionSet';
 
-// Intentionally does nothing. Implementation will be added in future PR.
-// eslint-disable-next-line no-unused-vars
-const observer = new MutationObserver(mutations => {
-  // These are only called so I can check what the module output looks like. The
-  // code is unreachable.
-  clientRenderBoundary();
-  completeBoundaryWithStyles();
-  completeBoundary();
-  completeSegment();
-});
+if (!window.$RC) {
+  // TODO: Eventually remove, we currently need to set these globals for
+  // compatibility with ReactDOMFizzInstructionSet
+  window.$RC = completeBoundary;
+  window.$RM = new Map();
+}
+
+if (document.readyState === 'loading') {
+  if (document.body != null) {
+    installFizzInstrObserver(document.body);
+  } else {
+    // body may not exist yet if the fizz runtime is sent in <head>
+    // (e.g. as a preinit resource)
+    const domBodyObserver = new MutationObserver(() => {
+      // We expect the body node to be stable once parsed / created
+      if (document.body) {
+        if (document.readyState === 'loading') {
```

Comment:
Not sure the order of DOM parsing / changing readyState - this could totally be a redundant check 

Added just in case readyState gets set `loading` -> `interactive` before mutation observer listeners are triggered.

(experimented with inline scripts on my laptop's version of Chrome, but wary about other browser implementations)

Reply:
I don't know off the top of my head. seems safe enough to leave in